src/states/node.rs

```diff
@@ -1426,6 +1426,17 @@ impl Node {
             }
         }
 
+        let acting_as_proxy = if let Authority::Client { ref proxy_node_name, .. } =
+                                     routing_msg.src {
+            proxy_node_name == self.name()
+        } else {
```

Comment:
I thought using a standalone variable named as `acting_as_proxy` helped self-document the check.  I could move the `!acting_as_proxy` check to the front of the `if` condition to allow it to short-circuit the rest of the conditions, which should achieve the same thing.  Would that be OK?